src/Order/OrderApplication/Carts/DeleteItem.cs

```diff
@@ -0,0 +1,41 @@
+using API.Abstraction.Helpers;
+using DAL;
+using DAL.Orders;
+using DAL.Orders.Repository;
+using MediatR;
+
+namespace OrderApplication.Carts
+{
+    public class DeleteItem
+    {
+        public class Command : IRequest<Result<Unit>>        
+        {
+            public Guid CartId { get; set; }
+            public long SeatId { get; set; }
+            public int EventId { get; set; }
+        }
+
+        public class RequestHandler : IRequestHandler<Command, Result<Unit>>
+        {
+            private readonly ICartItemRepository _cartItemsRepository;
+
+            public RequestHandler(ICartItemRepository cartItemsRepository)
+            {
+                _cartItemsRepository = cartItemsRepository;
+            }
+
+            public async Task<Result<Unit>> Handle(Command request, CancellationToken cancellationToken)
+            {
+                var cartItem = await _cartItemsRepository.GetBy(request.CartId, request.EventId, request.SeatId);
+                if (cartItem is null)
+                    return Result<Unit>.Failure($"Did not find item in cart '{request.CartId}' with event id '{request.EventId}' and seat id '{request.SeatId}'.");
+                await _cartItemsRepository.Delete(cartItem.Id);
+                var result = await _cartItemsRepository.Save();
```

Comment:
Shouldn't we return seat to available state here?

Reply:
Yeah, haven't thought about that at the time